### DISCUSSION TITLE
Overlapping Annotations combine colours to show a different colour

### DISCUSSION THREAD

**Discussion by meteorquake** [sumatrapdfreader/sumatrapdf#4743] [Q&A]
This has bugged me for a long time, because I use colours to mean specific things.
When an annotation is created over an existing annotation, where their colours overlap it changes to a completely different interacted colour.
As far as normal usage should go, applying a highlight where one already exists should not combine the two colours except as far as the alpha setting implies. At full opacity the one later in time/higher in z-index should fully obscure the one that is under it.
Is there any way to achieve this, since it is a vital thing...
Cheers, David

**Comment by GitHubRulesOK** [comment]
The problem is common to all annotations here are the ones done in Edge where 2 colours overlapping become 3 colours. the problem is more noticeable when converted to MuPDF highlights, depending on colour contrast of the blended colours.

![image](https://github.com/user-attachments/assets/9f0a8bf4-1aeb-4e0e-95eb-1170f38e91f2)
![image](https://github.com/user-attachments/assets/f16d1f24-bb72-49df-8328-84021c576d5a)
![image](https://github.com/user-attachments/assets/a28b09c2-d5fe-4164-a707-60d5499ff23f)

**Comment by meteorquake** [comment]
The example you give there is at times survivable (maybe not your end one!), however this sort of colour change (here with sumatra) is extremely problematic -
![a](https://github.com/user-attachments/assets/87294797-f6e0-44d2-89ba-db1adaee99a2)

**Reply by GitHubRulesOK** [reply]
overlapping red and yellow from my schooldays in art class generated a dirty orange and it has not changed in the past 70 odd years (even with more modern PDF.)

Adobe designed highlights to be seen through they float closest to viewer screen (no way to change that) so both colours blend as neither can be an opaque OVERLAY.
As I said will be similar in Edge (Acrobat enhanced) 
![image](https://github.com/user-attachments/assets/6319c281-601c-48d7-86bc-2391ef9cc7e1)

**Comment by meteorquake** [comment]
Well I learnt something new there - I've tried it in Foxit and as you say, just the same!
It seems to me a shame that presumably the fundamental renderer must not have some option to use the sequentially- or time-later item (annotations seem to have timestamps) as the combination colour when calculating the united overlay as presumably it would take scarcely any extra coding to do so. Ah well :)

**Reply by GitHubRulesOK** [reply]
All annotations are basically unaware of the page or other annotations.
They simply display in the order encountered so a whole page of yellow (Red and Green NO Blue) will mixed with a whole page of just Blue look like a page of Green (NO Red NO Blue !), whichever order they are placed over the underlying rendered  imagery. 

They do not have to be the same units as the page content thus "one pixel" of each colour can be set to cover part or the whole page.

![image](https://github.com/user-attachments/assets/66d03fa8-1f2d-4fe2-a71f-057eecb79981)
